pymatgen/command_line/bader_caller.py

```diff
@@ -193,3 +206,175 @@ def get_oxidation_state_decorated_structure(self):
                    for i in range(len(structure))]
         structure.add_oxidation_state_by_site(charges)
         return structure
+
+    @property
+    def result(self):
+
+        summary = {
+            "min_dist": [d['min_dist'] for d in self.data],
+            "charge": [d['charge'] for d in self.data],
+            "volume": [d['atomic_vol'] for d in self.data],
+            "vacuum_charge": self.vacuum_charge,
+            "vacuum_volume": self.vacuum_volume,
+            "reference_used": self.reference_used,
+            "bader_version": self.version,
+        }
+
+        if self.potcar:
+            charge_transfer = [self.get_charge_transfer(i) for i in range(len(self.data))]
+            summary['charge_transfer'] = charge_transfer
+
+        return BaderResult(summary)
+
+
+class BaderResult(MSONable):
```

Comment:
You can rename BaderAnalysis to BaderCaller. Just add a deprecation stub. 

Perhaps the question I have is why you need to make it MSONable. MSONAble is used when you have an object that cannot be easily serialized to json. A dict of list of values can already be serialized without additional effort.

Also, in general, I prefer if MSONable classes use the default as_dict and from_dict methods without subclass modification. This usually can be done quite easily. Implementing a custom as_dict and from_dict makes it very likely for people to add a new arg/kwarg and then forget to update the as_dict and from_dict methods. Following the typical MSONAble protocol automatically deals with this.